src/Functional.CQS.AOP.IoC.PureDI.MetricsCapturing.Tests/AsyncQueryHandlerMetricsCapturingDecoratorForUniversalStrategyTests.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using AutoFixture;
+using AutoFixture.Xunit2;
+using FakeItEasy;
+using Functional.CQS.AOP.CommonTestInfrastructure.DummyObjects;
+using Functional.CQS.AOP.MetricsCapturing;
+using Xunit;
+
+namespace Functional.CQS.AOP.IoC.PureDI.MetricsCapturing.Tests
+{
+	public class AsyncQueryHandlerMetricsCapturingDecoratorForUniversalStrategyTests
+	{
+		[Theory]
+		[AsyncQueryHandlerCompletesSuccessfully]
+		public async Task ShouldCaptureResultAndElapsedTime(
```

Comment:
This test name may be misleading; we only assert that `metricsCapturingStrategy.OnInvocationStart()` and the appropriate "completed" method _(`OnInvocationCompletedSuccessfully()` or `OnInvocationException()`)_ are called; _not_ the effects of those calls.

Reply:
Same comment applies to `AsyncQueryHandlerMetricsCapturingDecoratorTests`.

Reply:
Will be changing the test names for universal strategies from `ShouldCaptureResultAndElapsedTime` to `ShouldCaptureElapsedTime`, because universal strategies do not work with individual result types.